src/OpenSage.Game/DataStructures/Quadtree.cs

```diff
@@ -1,266 +1,60 @@
-        {
-            var strokeColor = new ColorRgbaF(0, 220, 0, 255);
+    public IEnumerable<T> FindIntersecting(T searcher) => Find(searcher.RoughCollider, searcher);
 
-            var ltWorld = new Vector3(Bounds.Position + new Vector2(0, Bounds.Height), 0);
-            var rtWorld = new Vector3(Bounds.Position + new Vector2(Bounds.Width, Bounds.Height), 0);
-            var rbWorld = new Vector3(Bounds.Position + new Vector2(Bounds.Width, 0), 0);
-            var lbWorld = new Vector3(Bounds.Position, 0);
+    public IEnumerable<T> FindIntersecting(in Collider collider, T searcher = null) => Find(collider, searcher);
 
-            var ltScreen = camera.WorldToScreenPoint(ltWorld).Vector2XY();
-            var rtScreen = camera.WorldToScreenPoint(rtWorld).Vector2XY();
-            var rbScreen = camera.WorldToScreenPoint(rbWorld).Vector2XY();
-            var lbScreen = camera.WorldToScreenPoint(lbWorld).Vector2XY();
+    public IEnumerable<T> Find(in Collider collider, T searcher, bool twoDimensional = false, bool hasToCollideWithSearcher = true)
+    {
+        return !collider.Intersects(Bounds) ? [] : FindIntersectingInternal(collider, searcher, twoDimensional, hasToCollideWithSearcher);
+    }
 
-            drawingContext.DrawLine(new Line2D(ltScreen, lbScreen), 1, strokeColor);
-            drawingContext.DrawLine(new Line2D(lbScreen, rbScreen), 1, strokeColor);
-            drawingContext.DrawLine(new Line2D(rbScreen, rtScreen), 1, strokeColor);
-            drawingContext.DrawLine(new Line2D(rtScreen, ltScreen), 1, strokeColor);
+    // collider is what we use to determine intersections _unless_ hasToCollideWithSearcher is true, in which case searcher is also used
+    // otherwise searcher is just excluded from the results
+    // twoDimensional is passed into CollidesWith calls
 
-            if (!IsLeaf)
-            {
-                foreach (var child in _children)
-                {
-                    child.DebugDraw(drawingContext, camera);
-                }
-            }
-        }
+    private IEnumerable<T> FindIntersectingInternal(Collider collider, T searcher, bool twoDimensional = false, bool hasToCollideWithSearcher = true)
+    {
+        return _rootNode.Find(collider, searcher, twoDimensional, hasToCollideWithSearcher);
+    }
 
-        internal enum Quad
-        {
-            UpperLeft = 0,
-            UpperRight,
-            LowerLeft,
-            LowerRight,
-        }
+    public bool Insert(in T item)
+    {
+        return _rootNode.Insert(item);
     }
 
-    public interface ICollidable
+    public bool Remove(in T item)
     {
-        Collider RoughCollider { get; }
-        List<Collider> Colliders { get; }
-        Vector3 Translation { get; }
+        return _rootNode.Remove(item);
+    }
 
-        bool CollidesWith(ICollidable other, bool twoDimensional);
+    public bool Update(in T item)
+    {
+        Remove(item);
+        return Insert(item);
```

Comment:
This was just copied over from the existing implementation, but there's probably room for optimization here. If we can determine that the node the item is in still fully captures the item, then we don't actually need to do anything here.